web/src/engine/websites/decorators/Common.ts

```diff
@@ -521,6 +522,35 @@ export function ImageElement(includeRefererHeader = true, detectMimeType = false
     };
 }
 
+export async function FetchImageAjaxFromHTML(page: Page, priority: Priority, signal: AbortSignal, queryImage: string, detectMimeType = false): Promise<Blob> {
+    const image = await this.imageTaskPool.Add(async () => {
+        const request = new FetchRequest(page.Link.href, {
+            signal: signal
```

Comment:
Should a referer header be added?
```typescript
headers: {
    Referer: page.Parameters?.Referer || page.Link.origin,
}
```

Reply:
I think the best option is to : 

1) Use page.Link.origin as referrer for getting page, always.
2) Use Page.parameters.Referrer OR (if it doenst exists) page.link as referer to pass in our custom fake page passed to FetchImageAjax